features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemTextView.kt

```diff
@@ -77,14 +77,22 @@ internal fun getTextWithResolvedMentions(content: TimelineItemTextBasedContent):
     val userProfileCache = LocalRoomMemberProfilesCache.current
     val lastCacheUpdate by userProfileCache.lastCacheUpdate.collectAsState()
     val mentionSpanTheme = LocalMentionSpanTheme.current
-    val formattedBody = remember(content.formattedBody, mentionSpanTheme, lastCacheUpdate) {
-        content.formattedBody?.let { formattedBody ->
-            updateMentionSpans(formattedBody, userProfileCache)
-            mentionSpanTheme.updateMentionStyles(formattedBody)
-            formattedBody
+    val textWithMentions = if (content.formattedBody != null) {
+        remember(content.formattedBody, mentionSpanTheme, lastCacheUpdate) {
+            content.formattedBody?.let { formattedBody ->
+                updateMentionSpans(formattedBody, userProfileCache)
+                mentionSpanTheme.updateMentionStyles(formattedBody)
+                formattedBody
+            }
+        }
+    } else {
+        remember(content.pillifiedBody, mentionSpanTheme, lastCacheUpdate) {
+            updateMentionSpans(content.pillifiedBody, userProfileCache)
+            mentionSpanTheme.updateMentionStyles(content.pillifiedBody)
+            content.pillifiedBody
```

Comment:
Is all this block equivalent to this simpler form?

```kotlin
    val formattedBody = content.formattedBody ?: content.pillifiedBody
    val textWithMentions = remember(formattedBody, mentionSpanTheme, lastCacheUpdate) {
        updateMentionSpans(formattedBody, userProfileCache)
        mentionSpanTheme.updateMentionStyles(formattedBody)
        formattedBody
    }
```

Reply:
Yeah, nice catch!